src/components/Calculator.js

```diff
@@ -1,30 +1,56 @@
 import '../index.css';
+import React, { useState } from 'react';
 import Calculator from './ButtonComponent';
+import calculate from '../logic/calculate';
 
 function CalculatorDesign() {
+  const initialState = {
+    total: null,
+    next: null,
+    operation: null,
+  };
+
+  const [state, setState] = useState(initialState);
+
+  // Array destructuring
+
+  const { total, operation, next } = state;
+
+  // button click event
+  const eventhandler = (event, name) => {
+    const updateState = calculate(state, name);
+    setState(updateState);
+  };
   return (
     <div className="calculator">
-      <div className="result">100</div>
+      <div className="result">
+        {total}
+        {' '}
+        {operation}
+        {' '}
+        {next}
+        {' '}
+      </div>
       <div className="keyboard">
-        <Calculator name="AC" className="nosign" />
-        <Calculator name="+/-" className="nosign" />
-        <Calculator name="%" className="nosign" />
-        <Calculator name="÷" className="side" />
-        <Calculator name="7" className="nosign" />
-        <Calculator name="8" className="nosign" />
-        <Calculator name="9" className="nosign" />
-        <Calculator name="*" className="side" />
-        <Calculator name="4" className="nosign" />
-        <Calculator name="5" className="nosign" />
-        <Calculator name="6" className="nosign" />
-        <Calculator name="-" className="side" />
-        <Calculator name="1" className="nosign" />
-        <Calculator name="2" className="nosign" />
-        <Calculator name="3" className="nosign" />
-        <Calculator name="+" className="side" />
-        <Calculator name="0" className="span" />
-        <Calculator name="." className="nosign" />
-        <Calculator name="=" className="side" />
+        <Calculator onClick={eventhandler} name="AC" className="nosign" />
+        <Calculator onClick={eventhandler} name="+/-" className="nosign" />
+        <Calculator onClick={eventhandler} name="%" className="nosign" />
+        <Calculator onClick={eventhandler} name="÷" className="side" />
+        <Calculator onClick={eventhandler} name="7" className="nosign" />
+        <Calculator onClick={eventhandler} name="8" className="nosign" />
+        <Calculator onClick={eventhandler} name="9" className="nosign" />
+        <Calculator onClick={eventhandler} name="*" className="side" />
```

Comment:
- [ ] Great effort working on the event milestone of the Math Magician project, But currently there is something you need to work on to make your calculation looks perfect and inline with the requirement. I notice when I `add, substract, divide` it all works fine but once I try to multiply a number it throw an `Unknown Operation '*'` error and it makes it difficult for user to multiply numbers. Kindly work on it and be sure all functionality is working perfectly.

Video Demonstration 👇 

[screen-capture.webm](https://github.com/rabayarabu/calculator/assets/78634841/bf838e18-2448-4dfa-a75d-3341a333fcdc)